include/libtorrent/aux_/vector.hpp

```diff
@@ -39,11 +39,22 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent { namespace aux {
 
+	template <typename IndexType>
+	struct vector_index_t
```

Comment:
I think this could be a lot simpler. something like this (I may get the syntax wrong):

template <typename T> struct underlying_type { using type = T; };
template <>
template<typename I> struct underlying_type<strong_typedef<I>> { using type = I; };